Sources/Sentry/Profiling/SentryProfiledTracerConcurrency.mm

```diff
@@ -351,6 +338,10 @@
                 @"Continuous profiling v1 configured; will not start automatically for trace.");
             return nil;
         }
+        if (transactionContext.parentSpanId != nil) {
+            SENTRY_LOG_DEBUG(@"Not a root span, will not start automatically for trace lifecycle.");
+            return nil;
+        }
```

Comment:
In some cases we were tracking child spans, checking for parents fixed issues related to that.